src/Utilities.ts

```diff
@@ -105,4 +105,36 @@ export class Utilities<TEightBittr extends FullScreenPokemon> extends GameStartr
 
         return true;
     }
+
+    /**
+     * Function to check if a pokemon should evolve at current level.
+     * 
+     * @param actor   Pokemon to be checked for evolution.
+     * @returns Whether the pokemon is able to evolve at current level.
+     */
+    public shouldEvolve(actor: IPokemon): boolean {
+        let t: string = actor.title.join("");
+        let p: any = this.EightBitter.MathDecider.getConstant("pokemon")[t];
```

Comment:
`p` definitely has a known type. `IPokemon` or `IPokemonListing` or something like that.